zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template/proposal.rs

```diff
@@ -42,23 +42,22 @@ pub enum ProposalResponse {
 
 impl ProposalResponse {
     /// Return a rejected response containing an error kind and detailed error info.
-    pub fn rejected<S: ToString>(kind: S, error: BoxError) -> Self {
-        let kind = kind.to_string();
-
-        // Pretty-print the detailed error for now
-        ProposalResponse::Rejected(format!("{kind}: {error:#?}"))
-    }
-
-    /// Return a rejected response containing just the detailed error information.
-    pub fn error(error: BoxError) -> Self {
-        // Pretty-print the detailed error for now
-        ProposalResponse::Rejected(format!("{error:#?}"))
-    }
-}
-
-impl<E: Error + Send + Sync + 'static> From<E> for ProposalResponse {
-    fn from(error: E) -> Self {
-        Self::error(error.into())
+    ///
+    /// Note: Error kind is currently ignored to match zcashd error format (`kebab-case` string).
+    pub fn rejected<S: ToString>(_kind: S, error: BoxError) -> Self {
+        // Make error `kebab-case` to match zcashd format.
+        let error_kebab1 = format!("{error:?}")
+            .replace(|c: char| !c.is_alphanumeric(), "-")
+            .to_ascii_lowercase();
+
+        // Remove consecutive `-` characters if any.
+        let regex1 = regex::Regex::new(r"-+").expect("valid regex");
+        let error_kebab2 = regex1.replace_all(&error_kebab1, "-");
+        // Remove trailing `-` character if any.
+        let regex2 = regex::Regex::new(r"-$").expect("valid regex");
```

Comment:
Optional/Nitpick: I don't think any of the errors that might be converted need to be checked for leading '-' characters, but in lieu of tests, could we check anyways?

```suggestion
        // Trim any leading or trailing `-` characters.
        let regex2 = regex::Regex::new(r"^-|-$").expect("valid regex");
```

Reply:
This will only trim a single leading or trailing character, using `trim_matches` would be more reliable here.

Reply:
Both suggestions are good to add, i made the `trim_matches` one at https://github.com/ZcashFoundation/zebra/pull/6044/commits/154386cc4ca94cfa9cff6b5fcce8eb6350556e33 so we have one regex less.